[PR port] Port #53823 - Enable master stats for tests

barneysowood opened the pull request:
PR port #53823
- [ ] Docs
- [X] Changelog
- [ ] Tests written/updated

Ch3LL commented:
@barneysowood would you mind resolving the merge conflicts?

barneysowood commented:
> @barneysowood would you mind resolving the merge conflicts?

@Ch3LL - no problem, I've done that now.

Ch3LL commented on salt/master.py:
```diff
@@ -1037,16 +1038,17 @@ def _post_stats(self, start, cmd):
             )
-            self.stats = collections.defaultdict(lambda: {"mean": 0, "runs": 0})
             self.stat_clock = end
+            log.error("STATS %s", pprint.pformat(stats_last))
```
I wonder if we want to error here. I know you weren't the original author so @dwoz do you still want to log error here. Seems a bit noisy

twangboy commented on salt/master.py:
```diff
@@ -9,6 +9,7 @@
 import logging
 import multiprocessing
 import os
+import pprint
```
This is probably your pre-commit failure... pprint is unused

dwoz committed:
Enable master stats for tests

Add master_stats: true to integration test master configuration so the
master statistics feature is exercised during integration test runs.

dwoz commented:
Rebased on current master and stripped the PR down to its functional core:

- Removed `log.error("STATS ...")` and `pprint` import (addressed @Ch3LL's review feedback)
- Removed `latency` field from stats defaultdict (was unused; not in the original #53823 scope)
- Added `master_stats: true` to integration test master conf (the original intent)
- Updated changelog to current format (`58099.changed.md`)

2 files changed, net +4 lines.
